spacemeshos/smapp
# Allow customisation of node config

It closes #1019 

**Additionally**: resolving an issue with stale genesisID for managers, because of [spawn managers](https://github.com/spacemeshos/smapp/pull/1248/files#diff-6647c75e9f02aabafdca57709814859558de581c5e713d87e8c8f57780fc2ee9R21), currently in develop, we're passing the genesisID only once and after it, reuse the genesisID and only change it on the wallet change or app restart. 

**UPD** How to test 

1. Test custom config 
 - run ./Spacemesh --discovery=http://127.0.0.1:8000/networks.json --checkInterval=30
 - pass the node setup process
 - take node-config.testnet-5.json and see smeshing options, take bootnodes with p2p option from the node-config.json , place in node-config.testnet-5.json and update it.
 - wait 30s , check node-config.json and see update 
  
 
2. Test update from discovery 

 - similar process, but update testnet-5.json config in the local network , update it, and see updates in node-config.json
2 case 
 - update the testnet-5.json network config, but also apply the same but different changes for the node-config.testnet-5.json file
 - it should take updates from custom config ( node-config.testnet-5.json ) and ignore testnet-5.json

3. Test smeshing updates 

- we run smapp 1 time and did not the init the PoS process.
- check the node-config.testnet-5.json file, should see smeshing options, with init post dir, only 1 
- go to smeshing screen and init PoS , and right now in the node-config.testnet-5.json  file , u will have PoS information, which will be applied to node-config.json file. 
- stop the smeshing process should clean up the smeshing options in node-config.testnet-5.json   and for node-config.json accordingly 

4. Test compatibility with latest develop without custom config
- go to release ( develop right now ) , set up PoS and node 
- back to my branch and run it
- settings from PoS ( from develop ) , should be migrated to my , and file cutoms config should be updated with this settings 

5. Test change network
1 . set up smeshing
2. go to the settings
3. choose different network
4. check the PoS screen and see the result, smeshing should not be inited 
5. init the smeshing on the PoS screen 
6 . switch back to the first network and compare.


https://github.com/spacemeshos/smapp/assets/54288612/f7e008e2-a37a-43e7-9b2e-1a606a6b20f5
